extensions/kafka-eight-simpleConsumer/src/main/java/io/druid/firehose/kafka/KafkaEightSimpleConsumerFirehoseFactory.java

```diff
@@ -0,0 +1,319 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.firehose.kafka;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.google.common.collect.Maps;
+import com.metamx.common.logger.Logger;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Sets;
+
+import io.druid.data.input.ByteBufferInputRowParser;
+import io.druid.data.input.Committer;
+import io.druid.data.input.FirehoseFactoryV2;
+import io.druid.data.input.FirehoseV2;
+import io.druid.data.input.InputRow;
+import io.druid.firehose.kafka.KafkaSimpleConsumer.BytesMessageWithOffset;
+
+public class KafkaEightSimpleConsumerFirehoseFactory implements
+    FirehoseFactoryV2<ByteBufferInputRowParser>
+{
+  private static final Logger log = new Logger(
+      KafkaEightSimpleConsumerFirehoseFactory.class
+  );
+
+  @JsonProperty
+  private final List<String> brokerList;
+
+  @JsonProperty
+  private final List<Integer> partitionIdList;
+
+  @JsonProperty
+  private final String clientId;
+
+  @JsonProperty
+  private final String feed;
+
+  @JsonProperty
+  private final int queueBufferLength;
+
+  @JsonProperty
+  private boolean earliest;
+
+  private final List<PartitionConsumerWorker> consumerWorkers = new CopyOnWriteArrayList<>();
+  private static final int DEFAULT_QUEUE_BUFFER_LENGTH = 20000;
+  private static final String RESET_TO_LATEST = "latest";
+  private static final int CONSUMER_FETCH_TIMEOUT = 10000;
+  @JsonCreator
+  public KafkaEightSimpleConsumerFirehoseFactory(
+      @JsonProperty("brokerList") List<String> brokerList,
+      @JsonProperty("partitionIdList") List<Integer> partitionIdList,
+      @JsonProperty("clientId") String clientId,
+      @JsonProperty("feed") String feed,
+      @JsonProperty("queueBufferLength") Integer queueBufferLength,
+      @JsonProperty("resetBehavior") String resetBehavior
+  )
+  {
+    this.brokerList = brokerList;
+    this.partitionIdList = partitionIdList;
+    this.clientId = clientId;
+    this.feed = feed;
+
+    this.queueBufferLength = queueBufferLength == null ? DEFAULT_QUEUE_BUFFER_LENGTH : queueBufferLength;
+    log.info("queueBufferLength loaded as[%s]", this.queueBufferLength);
+
+    this.earliest = RESET_TO_LATEST.equalsIgnoreCase(resetBehavior) ? false : true;
```

Comment:
If the reset is going to be treated as a boolean anyway, it might as well just be a boolean. That would prevent weirdness from happening when someone sets it to something other than "earliest" or "latest" and gets surprised when it magically turns into "earliest".